src/tmpfiles/tmpfiles.c

```diff
@@ -201,6 +201,11 @@ static int specifier_machine_id_safe(char specifier, void *data, void *userdata,
 static int specifier_user_name(char specifier, void *data, void *userdata, char **ret) {
         char *t;
 
+        if (!arg_user) {
+                log_error("%%U/%%u/%%h specifiers can only be used in --user mode");
+                return -EINVAL;
+        }
```

Comment:
hmm, i'd probably allow this, and make it behave exactly like in unit files, i.e. always resolve to `root`... Also allows us to share common code á la #7444